homeassistant/components/recorder/core.py

```diff
@@ -377,7 +377,8 @@ def _async_event_filter(self, event: Event) -> bool:
         # Unknown what it is.
         return True
 
-    def _empty_queue(self, event: Event) -> None:
+    @callback
+    def _async_empty_queue(self, event: Event) -> None:
```

Comment:
This event ended up being scheduled to run in a executor on shutdown. While it actually operated on standard async queue features. So it just delayed the shutdown step.

Reply:
`self._queue` is an `self._queue: queue.SimpleQueue[RecorderTask] = queue.SimpleQueue()`

It meant to be read in the recorder thread only but this is a failsafe to clear the queue and since it doesn't block and we are going to throw away the results, its fine to do it in the main thread

Reply:
It was in an executor thread before. Which is equally bad. I would assume it is somewhat thread safe. Since we add events too it from non recorder threads.